test/test-waitForText.js

```diff
@@ -10,9 +10,32 @@ buster.testCase('waitForText',
   // create new test object for each test
   setUp: common.setUp,
 
+  'Recursively calls itself if the element is not present after cooldown period': function()
+  {
+    var target      = {text: this.spy(), length: 0}
```

Comment:
If you pass a jQuery object and it's empty, it won't be updated automatically if matching element would appear on the page. This is why we need selectors, rather than objects.